File: src/main/java/myLang/block/print/PrintVariationsContext.java
package myLang.block.print;

import myLang.ContextHandler;
import myLang.expression.ExpressionContext;
import myLang.response.EmptyResponse;
import myLang.response.MyLangException;
import myLang.response.Response;
import myLangParser.MyLangLexer;
import myLangParser.MyLangParser;

public class PrintVariationsContext implements ContextHandler<MyLangParser.PrintVariationsContext> {
    @Override
    public Response handler(MyLangParser.PrintVariationsContext ctx) throws MyLangException {
        if(ctx.getChildCount() >= 3 && ctx.start.getType() == MyLangLexer.STRING && ctx.COLON() != null){
            Response response = new ExpressionContext().handler(ctx.expression());
            System.out.print(ctx.start.getText().replace("\"", "") + " " + response.getResponse());
        } else if ( ctx.start.getType() == MyLangLexer.STRING){
            System.out.print(ctx.start.getText().replace("\"", ""));
        } else if (ctx.expression() != null){
            Response response = new ExpressionContext().handler(ctx.expression());
            System.out.print(response.getResponse());
        } else {
            throw new MyLangException(ctx.getText() + "\nНереализованная форма оператора print");
        }

        if(ctx.printVariations().size() != 0){
            System.out.print(" ");
        } else {
            System.out.println("");
        }

        for(MyLangParser.PrintVariationsContext context: ctx.printVariations()){
            handler(context);
        }

        return new EmptyResponse();
    }
}

File: src/main/kotlin/myLangKotlin/block/print/PrintVariationsContext.kt
package myLangKotlin.block.print

import myLangKotlin.ContextHandler
import myLangKotlin.expression.ExpressionContext
import myLangKotlin.response.EmptyResponse
import myLangKotlin.response.MyLangException
import myLangKotlin.response.Response
import myLangParser.MyLangLexer
import myLangParser.MyLangParser

class PrintVariationsContext : ContextHandler<MyLangParser.PrintVariationsContext> {
    @Throws(MyLangException::class)
    override fun handler(ctx: MyLangParser.PrintVariationsContext): Response<*> {
        if (ctx.childCount >= 3 && ctx.start.type == MyLangLexer.STRING && ctx.COLON() != null) {
            val response = ExpressionContext().handler(ctx.expression())
            print(ctx.start.text.replace("\"", "") + " " + response.response)
        } else if (ctx.start.type == MyLangLexer.STRING) {
            print(ctx.start.text.replace("\"", ""))
        } else if (ctx.expression() != null) {
            val response = ExpressionContext().handler(ctx.expression())
            print(response.response)
        } else {
            throw MyLangException(ctx.text + "\nНереализованная форма оператора print")
        }

        if (ctx.printVariations().size != 0) {
            print(" ")
        } else {
            println("")
        }

        for (context in ctx.printVariations()) {
            handler(context)
        }

        return EmptyResponse()
    }
}
